File: fizzbuzz.py
#=====================================#
#             FizzBuzz                #
#          by Randall Arms            #
#  @ github.com/randallarms/fizzbuzz  #
#=====================================#

# Opening text
print("\n\n=========")
print("FIZZBUZZ!")
print("=========")

# Checks that the input n is a positive integer and returns an obj of int type
def int_check(n):
    try:
        return int(n)
    except ValueError:
        print("\nInput number must be a positive integer. ")
        exit()

# Input the maximum integer to iterate to
print("\nUp to which number would you like to iterate? ")
i = input("> ");

i_int = int_check(i)

# Check for large numbers and confirm
if int(i) > 1000:
    print("\nSet is greater than 1000 and may take a long time to finish. Type yes to confirm: ")
    confirm = input("> ");
    if confirm != "yes":
	    print("\nIteration canceled. ")
	    exit()

# Input "Fizz" divisor
print("\nWhich number would you like to check divisibility by for printing 'Fizz'? ")
f = input("> ");

f_int = int_check(f)

# Input "Buzz" divisor
print("\nWhich number would you like to check divisibility by for printing 'Buzz'? ")
b = input("> ");

b_int = int_check(b)

# Iterate through the integers and print results
print("\nIterating through each integer up to " + i + ": ")
for num in range(1, i_int+1):
	fizz = (num % f_int == 0)
	buzz = (num % b_int == 0)
	print('FizzBuzz' if (fizz & buzz) else 'Fizz' if fizz else 'Buzz' if buzz else num)

File: FizzBuzz.java
/*===================================*\
#          FizzBuzz (Java)            #
#          by Randall Arms            #
#  @ github.com/randallarms/fizzbuzz  #
\*===================================*/

import java.util.Scanner;
import java.util.stream.IntStream;

public class FizzBuzz {
	
	// Checks that the input n is a positive integer and returns an obj of int type
	static int int_check(String n) {
		
		int c_int = 0;
		
		try {
			c_int = Integer.parseInt(n);
		} catch(NumberFormatException e) {
			System.out.println("\nInput number must be an integer. ");
			System.exit(0);
		}
		
		if (c_int >= 0) {
			return c_int;
		} else {
			System.out.println("\nInput number must be a positive integer. ");
			System.exit(0);
			return 0;
		}
		
	}
	
	// Main method
	public static void main(String[] args) {

		// Opening text
		System.out.println("\n\n=========");
		System.out.println("FIZZBUZZ!");
		System.out.println("=========");
		
		// Scanner obj
		Scanner scanner = new Scanner(System.in);

		// Input the maximum integer to iterate to
		System.out.println("\nUp to which number would you like to iterate? ");
		String i = scanner.nextLine();

		int i_int = int_check(i);

		// Check for large numbers and confirm
		if (i_int > 1000) {
			System.out.println("\nSet is greater than 1000 and may take a long time to finish. Enter 'yes' to confirm: ");
			String confirm = scanner.nextLine();
			if (!confirm.contains("yes")) {
				System.out.println("\nIteration canceled. ");
				System.exit(0);
			}
		}

		// Input "Fizz" divisor
		System.out.println("\nWhich number would you like to check divisibility by for printing 'Fizz'? ");
		String f = scanner.nextLine();

		int f_int = int_check(f);

		// Input "Buzz" divisor
		System.out.println("\nWhich number would you like to check divisibility by for printing 'Buzz'? ");
		String b = scanner.nextLine();

		int b_int = int_check(b);

		// Iterate through the integers and print results
		System.out.println("\nIterating through each integer up to " + i + ": ");
		IntStream range = IntStream.range(1, i_int+1);
		for (int num : range.toArray()) {
			boolean fizz = (num % f_int == 0);
			boolean buzz = (num % b_int == 0);
			System.out.println((fizz && buzz) ? "FizzBuzz" : fizz ? "Fizz" : buzz ? "Buzz" : Integer.toString(num));
		}
			
	}
	
}